feat: Added api services in pokedex

NicoEnriquezFFuFm opened the pull request:
- added the handler directory
- added api_client.dart
- added api_service.dart
- added pokemon_api.dart inside handler directory
- added model directory

NicoEnriquezFFuFm committed:
feat: Implemented api services in pokedex
- added the handler directory
- added api_client.dart
- added api_service.dart
- added pokemon_api.dart inside handler directory
- added model directory

shawn-ffufm commented on lib/api/api_service.dart:
```diff
@@ -0,0 +1,18 @@
+import 'package:pokedexbootcamp/api/handler/pokemon_api.dart';
+
+class ApiService {
+  static final ApiService _apiService = ApiService._internal(); //tells that its single use
```
do you still need this comment here? 🙁

shawn-ffufm commented on lib/api/api_service.dart:
```diff
@@ -0,0 +1,18 @@
+class ApiService {
+  static final ApiService _apiService = ApiService._internal(); //tells that its single use
+
+  factory ApiService() => _apiService; //global point reuse
```
same here

shawn-ffufm reviewed (changes requested):
@NicoEnriquezFFuFm see minor pr feedback 🙂

NicoEnriquezFFuFm committed:
fix: addressed pr feedback
- removed comment on line 5 inside api_service
- removed comment on line 7 inside api_service
